webapp/packages/game/src/index.ts

```diff
@@ -10,6 +10,7 @@ import goalCards from './data/card/goals.json';
 import { isInRange, zip } from './utils';
 
 type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+type WithGameState<G extends any, F extends (...args: any) => void> = (G: State<G>['G'], ctx: State<G>['ctx'], ...args: Parameters<F>) => any;
```

Comment:
the reason why replace the default type `Move` with the customized type `WithGameState` is because of the generic type `args`?

Reply:
The `WithGameState` wrapper is helpful to declare the move method type once and shared it on both game core and clientside. The move parameters contain `G` and `ctx` in game core i.e. `move(G, ctx, param1, param2 /* and so on */)` but not in client side. i.e. `move(param1, param2 /* and so on */)`. Having this wrapper is helpful that we define move function type once and reused it across the platforms.